pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/AggregationExecutor.java

```diff
@@ -38,10 +38,9 @@
    * Asserts that 'init' has been called before calling this method.
    *
    * @param docIdSet
-   * @param startIndex
    * @param length
    */
-  void aggregate(int[] docIdSet, int startIndex, int length);
```

Comment:
The issue is that the checked in code only works when startIndex=0 (people already assumed that), which is always true. I looked at the code, this kind of API comes from the Dictionary APIs, like readDoubleValues(int[] dictionaryIds, int startPos, int limit, double[] outValues, int outStartPos). There are very little usage for these APIs, and all of them with (dictIds, 0, length, outValues, 0). Adding startPos and outStartPos will add more complexity to the code, and potentially hurt the performance. My suggestion is just remove the startPos and outStartPos, since no one actually need them.